pinot-common/src/main/java/org/apache/pinot/common/config/TableConfig.java

```diff
@@ -82,6 +85,10 @@
   @NestedConfig
   private RoutingConfig _routingConfig;
 
+  @ConfigKey(INSTANCE_ASSIGNMENT_CONFIG_MAP_KEY)
+  @UseChildKeyHandler(InstanceAssignmentConfigMapChildKeyHandler.class)
+  private Map<InstancePartitionsType, InstanceAssignmentConfig> _instanceAssignmentConfigMap;
```

Comment:
Map will be the best approach? We should consider other options to make this simpler. (e.g. put `instanceAssignmentConfigMap` + override config. `TenantConfig` is also using overriding config approach.

Reply:
After consideration, I think explicitly putting the type (OFFLINE/CONSUMING/COMPLETED) is more clear than using the override. For realtime segments, if COMPLETED config does not exist, use CONSUMING config instead

Reply:
In that case, do you think that we need to wrap those 3 config into some other class? It seems that adding 3 similar configs to the top level table config seems not too clean.